src/store/index.ts

```diff
@@ -80,41 +90,103 @@ export default createStore<State>({
         payload: messageData.ciphertext
       });
     },
-    async uploadFile({ dispatch }, request: UploadRequest) {
+    async uploadFile({ dispatch, getters, state }, request: UploadRequest) {
       const res = await axios.post<UploadResponse>(
         "https://rtc-data.registree.io/data/uploads",
         request
       );
-      if (res.status !== 200) {
+      const enclavePubKey = getters.enclavePublicKey;
+      const ourSecretKey = state.ourSecretKey;
+
+      if (res.status !== 200 || !enclavePubKey || !ourSecretKey) {
+        // TODO: Error Handling
+        return "error";
+      }
+
+      const { nonce, ciphertext } = res.data;
+
+      const msg = decryptMessage(
+        ciphertext,
+        nonce,
+        enclavePubKey,
+        ourSecretKey
+      );
+
+      if (!msg) {
+        // TODO: Error Handling
         return "error";
       }
-      await dispatch("decryptUploadResponse", res.data);
+
+      await dispatch("parseUploadMessage", msg);
     },
-    async decryptUploadResponse(
-      { state, dispatch, getters },
-      response: UploadResponse
-    ) {
+    async postAccessForm({ commit, getters, state }, request: UploadRequest) {
```

Comment:
It's probably worth making all the action return types explicit, to avoid accidental returns.

```suggestion
    async postAccessForm(
      { commit, getters, state },
      request: UploadRequest
    ): Promise<void> {
```